CribblyBackend/Services/TeamService.cs

```diff
@@ -0,0 +1,86 @@
+using System.Data;
+using System.Linq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using CribblyBackend.Models;
+using Dapper;
+
+namespace CribblyBackend.Services
+{
+    public interface ITeamService
+    {
+        Task<Team> GetById(int Id);
+        void Update(Team Team);
+        Task Create(Team Team);
+        void Delete(Team Team);
+    }
+    public class TeamService : ITeamService
+    {
+        IDbConnection connection;
+        public TeamService(IDbConnection connection)
+        {
+            this.connection = connection;
+        }
+
+        public async Task Create(Team team)
+        {
+            await connection.ExecuteAsync(
+                @"INSERT INTO Teams(Name) VALUES (@Name)", 
+                new { Name = team.Name }
+            );
+            foreach (Player player in team.Players)
+            {
+                await connection.ExecuteAsync(
+                    @"UPDATE Players SET TeamId = LAST_INSERT_ID() WHERE Id = @PlayerId", 
+                    new { PlayerId = player.Id });
+            }
+        }
+
+        public void Delete(Team team)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public async Task<Team> GetById(int id)
+        {
+            List<Player> playerList = new List<Player>();
+            //This returns two Team records, with each player showing up as the sole element of each Teams' Player List<Player>
+            var teamRecords = await connection.QueryAsync<Team, Player, Team>(
+                @"SELECT t.*, p.* FROM Teams t INNER JOIN Players p ON t.Id = p.TeamId WHERE p.TeamId = @Id", 
+                MapPlayerToTeams, 
+                new { Id = id },
+                splitOn: "Id"
+            );
+            //Add the Player List<Player> property from every record into a blank list
+            foreach (Team team in teamRecords)
+            {
+                playerList.AddRange(team.Players);
+            }
+            /*
+                We only need one copy of the Team object properties, so get the FirstOrDefault 
+                and overwrite the Player property with what we grabbed before
+            */
+            Team teamToReturn = teamRecords.FirstOrDefault();
```

Comment:
This works, but I think you can make it simpler:
```csharp
var players = new Dictionary<int, Player>();
var team = await connection.QueryAsync<Team, Player, Team>(
    @"SELECT t.*, p.* FROM Teams t INNER JOIN Players p ON t.Id = p.TeamId WHERE p.TeamId = @Id", 
    (p, t) =>
    {
        if (!players.TryGetValue(p.Id, out Player _))
        {
            players.Add(p);
        }
    }, 
    new { Id = id },
    splitOn: "Id"
    ).Distinct();
team.Players = players.Values.ToList();
```

Reply:
Also, your query uses `INNER JOIN`. Do you expect to be able to create teams with no players (perhaps an admin would do that)? If a team exists but there are no players with their team ID set to the team you're looking for, the inner join will return 0 rows (even though the team exists). If we want to allow teams with no players, we'd want to switch this to a left join.

Reply:
Unfortunately your code isn't working. It yells at me due to nothing being returned, however it doesn't like the Distinct() tag at the end when that happens. Changing INNER to LEFT works fine though

Reply:
Ope, yep, here:
```csharp
var team = (await connection.QueryAsync<Team, Player, Team>(
      @"SELECT t.*, p.* FROM Teams t INNER JOIN Players p ON t.Id = p.TeamId WHERE p.TeamId = @Id",
      MapPlayerToTeams,
      new { Id = id },
      splitOn: "Id"
  )).FirstOrDefault();
```
That should give you one team, then do the normal thing to populate the players.

Reply:
Except replace MapPlayersToTeams with the new thing